internal/provider/clickhouse_cluster_resource.go

```diff
@@ -643,6 +672,21 @@ func updateClickhouseCluster(m *clickhouseClusterModel) (*clickhouse.UpdateClust
 		rq.Access = access
 	}
 
+	cc := m.CustomCertificate.Attributes()
+	rq.CustomCertificate = &clickhouse.CustomCertificate{
+		Enabled: false,
+	}
+	certificate, certOk := cc["certificate"]
+	key, keyOk := cc["key"]
+	rq.CustomCertificate.Enabled = certOk && keyOk
+	if rq.CustomCertificate.Enabled {
```

Comment:
It's better to write dedicated `convert` method to hide this logic to clean function, which can be tested separately.
Otherwise, this method will be overburdened by too many complex lines.